sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js

```diff
@@ -0,0 +1,108 @@
+/** @odoo-module **/
+const {useState} = owl;
+import {BooleanField} from "@web/views/fields/boolean/boolean_field";
+import {_lt} from "@web/core/l10n/translation";
+import {registry} from "@web/core/registry";
+
+const iconTrue = "fa-check-square-o";
+const iconFalse = "fa-square-o";
+const tooltipTrue = "Switch to hidden";
+const tooltipFalse = "Switch to show";
+
+export class BooleanFaIconWidget extends BooleanField {
+    setup() {
+        super.setup();
+        this.state = useState({
+            fa_class: this.getFaIconClass(this.props.value),
+            text_tooltip: this.getTextTooltip(this.props.value),
+        });
+    }
+
+    getFaIconClass(currentValue) {
+        var fa_icons = this.props.fa_icons;
+        var icon_true = fa_icons.icon_true || iconTrue;
+        var icon_false = fa_icons.icon_false || iconFalse;
+        return currentValue ? icon_true : icon_false;
+    }
+
+    getTextTooltip(currentValue) {
+        var show_tooltip = this.props.terminology;
+        var tooltip_true = show_tooltip.hover_true || tooltipTrue;
+        var tooltip_false = show_tooltip.hover_false || tooltipFalse;
+        return currentValue ? _lt(tooltip_true) : _lt(tooltip_false);
+    }
+
+    isAllowEdit() {
+        var resModel = this.props.record.resModel;
+        var state = this.props.record.data.state;
+        if (typeof state === "undefined" || state === null) {
+            state = this.props.record.data.parent_state;
+        }
+        if (
+            ["account.move.line", "sale.order.line"].includes(resModel) &&
+            state !== "draft"
+        ) {
+            this.props.allow = false;
+            this.props.readonly = true;
+        }
+        return this.props.allow;
+    }
+
+    get isReadonly() {
+        return this.props.record.isReadonly(this.props.name);
+    }
+
+    get allow() {
+        return this.props.allow;
+    }
+
+    get fa_icons() {
+        return this.props.fa_icons;
+    }
+
+    get terminology() {
+        return this.props.terminology;
+    }
+
+    onChange(newValue) {
+        if (this.props.readonly || !this.isAllowEdit()) {
+            return;
+        }
+        super.onChange(newValue);
+        this.state.fa_class = this.getFaIconClass(newValue);
+        this.state.text_tooltip = this.getTextTooltip(newValue);
+    }
```

Comment:
```suggestion
    onClickButton(ev) {
        ev.stopPropagation();
        if (this.props.readonly || !this.isAllowEdit()) {
            return;
        }
        const newValue = !this.props.value
        this.props.update(newValue);
        this.state.fa_class = this.getFaIconClass(newValue);
        this.state.text_tooltip = this.getTextTooltip(newValue);
    }
```

Reply:
Thanks for the tips @CarlosRoca13 ,the issue was solved.